flow/layers/physical_shape_layer.cc

```diff
@@ -74,6 +75,15 @@ void PhysicalShapeLayer::Preroll(PrerollContext* context,
   }
 
   set_paint_bounds(paint_bounds);
+
+  if (UsesSaveLayer()) {
+    if (render_count_ >= kMinimumRendersBeforeCachingLayer) {
```

Comment:
PhysicalShapeLayer is legacy and deprecated. The framework no longer uses these layers.

See https://github.com/flutter/flutter/pull/102274

Reply:
I see, changes of PhysicalShapeLayer has been reverted.